### PR TITLE
limits.h: Map _POSIX_[S]SIZE_M[AX|IN] to [U]LONG_M[AX|IN]

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

correct SIZE_MAX and SSIZE_MAX/SSIZE_MIN definition

## Impact

SSIZE_MAX and SSIZE_MIN

## Testing

CI

### CODE REVIEW

File: include/limits.h [hunk 1]
```diff
@@ -136,11 +136,11 @@
 
 #else /* CONFIG_SMALL_MEMORY */
 
-#define _POSIX_SIZE_MAX       4294967295UL /* See sys/types.h */
+#define _POSIX_SIZE_MAX       ULONG_MAX /* See sys/types.h */
```
Comment (pkarashchenko) [inline include/limits.h:139]:
```suggestion
#define _POSIX_SIZE_MAX       ULONG_MAX
```

File: include/limits.h [hunk 2]
```diff
@@ -136,11 +136,11 @@
...
 #define _POSIX_SIZE_MIN       0
 
-#define _POSIX_SSIZE_MAX      2147483647L  /* See sys/types.h */
-#define _POSIX_SSIZE_MIN      -2147483648L
+#define _POSIX_SSIZE_MAX      LONG_MAX  /* See sys/types.h */
```
Comment (pkarashchenko) [inline include/limits.h:142]:
```suggestion
#define _POSIX_SSIZE_MAX      LONG_MAX
```

**Review by pkarashchenko — COMMENTED** [review]
Need to remove a comment since both reference values are defined in scope of the same file

**Review by acassis — APPROVED** [review]

File: include/limits.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/limits.h]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**